parlai/agents/transformer/image_polyencoder.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/env python3
+# Copyright (c) Facebook, Inc. and its affiliates.
+# This source code is licensed under the MIT license found in the
+# LICENSE file in the root directory of this source tree.
+
+# hack to make sure -m transformer/generator works as expected
+"""
+Poly-encoder agent that ingests image features.
+"""
+
+from typing import Any, Dict
+
+import torch
+
+from parlai.agents.image_seq2seq.modules import ContextWithImageEncoder
+from parlai.agents.transformer.modules import get_n_positions_from_options
+from parlai.agents.transformer.polyencoder import PolyencoderAgent, PolyEncoderModule
+from parlai.core.torch_agent import Batch
+from parlai.core.torch_image_agent import TorchImageAgent
+
+
+class ImagePolyencoderAgent(PolyencoderAgent, TorchImageAgent):
+    """
+    Poly-encoder Agent that ingests image features.
+
+    Agent that allows encoding image features and adding or concatenating them to the
+    context encoding.
+    """
+
+    @classmethod
+    def add_cmdline_args(cls, argparser):
+        """
+        Add command-line arguments specifically for this agent.
+        """
+        PolyencoderAgent.add_cmdline_args(argparser)
+        TorchImageAgent.add_cmdline_args(argparser)
+        agent = argparser.add_argument_group('Image Encoder Args')
+        agent.add_argument(
+            '--image-combination-mode',
+            type=str,
+            default='prepend',
+            choices=['add', 'append', 'prepend'],
```

Comment:
once we have a solid grasp of which method is best, let's add a `recommended='<method>'` flag here (could just mark as `TODO` for now)

Reply:
Added TODO for this